app/upgrades/v17/erc20.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package v17
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authzkeeper "github.com/cosmos/cosmos-sdk/x/authz/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/evmos/evmos/v16/precompiles/erc20"
+	erc20keeper "github.com/evmos/evmos/v16/x/erc20/keeper"
+	"github.com/evmos/evmos/v16/x/erc20/types"
+	evmkeeper "github.com/evmos/evmos/v16/x/evm/keeper"
+	transferkeeper "github.com/evmos/evmos/v16/x/ibc/transfer/keeper"
+)
+
+// RegisterERC20Extensions registers the ERC20 precompiles with the EVM.
+func RegisterERC20Extensions(
+	ctx sdk.Context,
+	authzKeeper authzkeeper.Keeper,
+	bankKeeper bankkeeper.Keeper,
+	erc20Keeper erc20keeper.Keeper,
+	evmKeeper *evmkeeper.Keeper,
+	transferKeeper transferkeeper.Keeper,
+) error {
+	precompiles := make([]vm.PrecompiledContract, 0)
+
+	var err error
+	erc20Keeper.IterateTokenPairs(ctx, func(tokenPair types.TokenPair) bool {
+		// skip registration if token is native or if it has already been registered
+		// NOTE: this should handle failure during the selfdestruct
+		if !tokenPair.IsNativeCoin() ||
```

Comment:
This is returning false if the tokenPair is not native. Is it correct?

Reply:
yes it just means that we continue the iteration -> a return value of `true` would stop the iteration 👍